Source/WebCore/Modules/WebGPU/GPUExternalTextureDescriptor.h

```diff
@@ -45,43 +45,33 @@ using GPUVideoSource = RefPtr<HTMLVideoElement>;
 
 struct GPUExternalTextureDescriptor : public GPUObjectDescriptorBase {
 
-    static PAL::WebGPU::VideoSourceIdentifier mediaIdentifierForSource(const GPUVideoSource& videoSource, CVPixelBufferRef& outPixelBuffer)
+    static std::pair<uint64_t, RefPtr<PAL::VideoFrame>> mediaIdentifierForSource(const GPUVideoSource& videoSource)
     {
 #if ENABLE(WEB_CODECS)
-        return WTF::switchOn(videoSource, [] (const RefPtr<HTMLVideoElement> videoElement) -> PAL::WebGPU::VideoSourceIdentifier {
+        return WTF::switchOn(videoSource, [](const RefPtr<HTMLVideoElement> videoElement) -> std::pair<uint64_t, RefPtr<PAL::VideoFrame>> {
             auto playerIdentifier = videoElement->playerIdentifier();
-            return PAL::WebGPU::HTMLVideoElementIdentifier { playerIdentifier ? playerIdentifier->toUInt64() : 0 };
+            return std::make_pair(playerIdentifier ? playerIdentifier->toUInt64() : 0, nullptr);
         }
-        , [&outPixelBuffer] (const RefPtr<WebCodecsVideoFrame> videoFrame) -> PAL::WebGPU::VideoSourceIdentifier {
-#if PLATFORM(COCOA)
-            if (auto internalFrame = videoFrame->internalFrame()) {
-                if (internalFrame->isRemoteProxy())
-                    return PAL::WebGPU::WebCodecsVideoFrameIdentifier { internalFrame->resourceIdentifier() };
+        , [](const RefPtr<WebCodecsVideoFrame> videoFrame) -> std::pair<uint64_t, RefPtr<PAL::VideoFrame>> {
+            if (auto internalFrame = videoFrame->internalFrame())
```

Comment:
This deals with "remote" logic.
The remote logic is not part of WebCore, WebCore doesn't understand that.
This should be: 
- invoked polymorphically with two implementations: one in webcore (in-process), one in webkit (gpup)
- outcome should be PAL::WebGPU::ExternalTextureDescriptor contents that makes sense to be used polymorphically by the receiver of the object

HOWEVER:: due to the design mistake of WebGPU using PAL, this cannot be such, as PAL doesn't have anything to build the polymorphism with. We cannot pass WebCore types polymorphically in non-type-punning manner.

You are end up smuggling the polymorphic data with upcasting.
For this purpose, you should have some sort of union with void* to avoid confusions and contraptions.